### PR TITLE
feat: add notifications UI components to be used by Views

### PR DESCRIPTION
**Pull request by Jonathansoufer**
## **Description**

This PR implements the UI pieces (components) that will serve as base layer for Notification's Views/Screen. At this stage,  although we're dealing with UI, there's no UI change sort of saying because the components aren't in use by any screen. The screen and final piece of this "break down" effort of PRs will bring all pieces together.

Main PR and visual final result can be seen [here](https://github.com/MetaMask/metamask-mobile/pull/9673)

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by Prithpal-Sooriya — DISMISSED** [review]

**Comment by tommasini** [comment]
@Jonathansoufer it's seems a lot e2e are failing, can you see what is happening?

File: app/components/UI/Notification/List/index.tsx [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+      const listProps: FlatListProps<Notification> = {
+        keyExtractor: (item) => item.id,
+        data,
+        ListEmptyComponent: (
+          <Empty
+            testID={NotificationsViewSelectorsIDs.NO_NOTIFICATIONS_CONTAINER}
+          />
+        ),
+        contentContainerStyle: styles.list,
+        // eslint-disable-next-line react/display-name, react/prop-types
```
Comment (tommasini) [inline app/components/UI/Notification/List/index.tsx:115]: I see prop types and display name being ignored, and I'm not understanding why we need to ignore it!

Also I think the ignore of prop-types could be called just once at the beginning of the component file

Comment (Jonathansoufer) [reply app/components/UI/Notification/List/index.tsx]: removed

**Review by Prithpal-Sooriya — APPROVED** [review]

**Review by tommasini — APPROVED** [review]
LGTM!

**Pull request merged**